src/main/java/com/google/idea/perf/TracerController.kt

```diff
@@ -132,66 +132,66 @@ class TracerController(
-                it.javaClass.getMethod(baseMethod.name, *baseMethod.parameterTypes)
+        val command = parseTracerCommand(cmd)
+
+        when (command) {
+            is TracerCommand.Clear -> {
+                callTree.clear()
+                updateTracepointUi()
             }
-            traceAndRetransform(*methods.toTypedArray())
-        }
-        else if (cmd.startsWith("trace")) {
-            val spec = cmd.substringAfter("trace").trim()
-            val split = spec.split('#')
-            if (split.size != 2) {
-                LOG.warn("Invalid trace request format; expected com.example.Class#method")
-                return
+            is TracerCommand.Reset -> {
+                callTree = MutableCallTree(Tracepoint.ROOT)
+                updateTracepointUi()
             }
-            val (className, methodName) = split
-            val classJvmName = className.replace('.', '/')
-            TracerConfig.traceMethods(classJvmName, methodName)
-            retransformClasses(setOf(className))
-        }
-        else if (cmd.startsWith("untrace")) {
-            val spec = cmd.substringAfter("untrace").trim()
-            val split = spec.split('#')
-            if (split.size != 2) {
-                LOG.warn("Invalid untrace format; expected com.example.Class#method")
-                return
+            is TracerCommand.Trace -> {
+                when (command.target) {
+                    is TraceTarget.PsiFinders -> {
+                        val defaultProject = ProjectManager.getInstance().defaultProject
+                        val psiFinders = PsiElementFinder.EP.getExtensions(defaultProject)
+                        val baseMethod = PsiElementFinder::findClass.javaMethod!!
+                        val methods = psiFinders.map {
+                            it.javaClass.getMethod(baseMethod.name, *baseMethod.parameterTypes)
+                        }
+                        if (command.enable) {
+                            traceAndRetransform(*methods.toTypedArray())
+                        }
+                        else {
+                            untraceAndRetransform(*methods.toTypedArray())
+                        }
+                    }
+                    is TraceTarget.Tracer -> {
+                        if (command.enable) {
+                            traceAndRetransform(
+                                TracerController::dataRefreshLoop.javaMethod!!,
+                                TracerController::updateTracepointUi.javaMethod!!,
+                                TracerController::handleCommand.javaMethod!!
+                            )
+                        }
+                        else {
+                            untraceAndRetransform(
+                                TracerController::dataRefreshLoop.javaMethod!!,
+                                TracerController::updateTracepointUi.javaMethod!!,
+                                TracerController::handleCommand.javaMethod!!
```

Comment:
This duplicates the list of method above. Please factor out a shared variable.
Note: you can use `arrayOf()` to declare the variable, and then use the spread operator (`*`) to forward the array elements as arguments to the `traceAndRetransform` function.